Lib/smtplib.py

```diff
@@ -1030,13 +1032,12 @@ def __init__(self, host='', port=0, local_hostname=None,
                                                      keyfile=keyfile)
             self.context = context
             SMTP.__init__(self, host, port, local_hostname, timeout,
-                    source_address)
+                          source_address)
 
         def _get_socket(self, host, port, timeout):
             if self.debuglevel > 0:
                 self._print_debug('connect:', (host, port))
-            new_socket = socket.create_connection((host, port), timeout,
-                    self.source_address)
```

Comment:
Wait, where is self.source_address gone? And why tests didn't catch that self.source_address is lost?

Reply:
@vstinner 
self.source_address was not passed as parameter, do you want to pass it as parameter?
https://github.com/python/cpython/blob/29cd207bee95b6598250e28cdf133ba4a068008a/Lib/smtplib.py#L311

Reply:
Oh wait, you replaced create_connecton() with _get_socket(). I misread the code. It is correct.